src/services/withHost.ts

```diff
@@ -0,0 +1,19 @@
+import { isUrl, Url } from '@/types/url'
+import { stringHasValue } from '@/utilities'
+
+export function withHost(url: Url, host?: string): Url
```

Comment:
Interesting that this is the only "with" utility where the second argument is optional. Why is that?

Reply:
agreed. I went through all the "with" utilities and made them much more similar. The only one that's different now is `withQuery` because it's also used in `createRouterResolve`. 

All of them now always return `Url`, host is sent as `string`, they all have an optional 2nd argument